src/slacko.ml

```diff
@@ -584,10 +584,13 @@ type history_result = [
   | timestamp_error
 ]
 
-let base_url = "https://slack.com/api/"
+let base_url = ref "https://slack.com/api/"
```

Comment:
I don't really like the thought of tying state to a module, because it makes it harder to (for example) talk to multiple slacks with different tokens.

What about something like this:
```ocaml
type session = {
  base_url : string;
  token : token;
  mutable cache : some_suitable_type; (* maybe the value itself can be mutable instead *)
}

let create_session ?(base_url="https://slack.com/api/") token = {
  base_url;
  token = token_of_string token;
  cache = empty_cache; (* or something similar *)
}
```

We make the `session` type abstract and we use it everywhere we're currently using `token`. We can also turn `token_of_string` into a wrapper that creates a new session and keep the external API the same for everything except functions that talk to slack and don't currently need a token (which I believe is only `api_test` and `oauth_access`).

I'm happy to prototype both this and the functor version you suggested (in separate branches) and see which one works better.